### PR TITLE
Quote programs in the bootstrap script

### PR DESCRIPTION
**Pull request by Chris00**
Fixes https://github.com/janestreet/jbuilder/issues/322

### CODE REVIEW

**Comment by Chris00** [comment]
Compilation with spaces in the program names: https://ci.appveyor.com/project/Chris00/ocaml-appveyor/build/1.0.198#L4631

**Comment by rgrinberg** [comment]
I'll leave all this tough quoting stuff to you @dra27 :). Let's make sure the CLA is present before merging though.

**Comment by Chris00** [comment]
> Let's make sure the CLA is present before merging though.

It is.

File: bootstrap.ml [hunk 1]
```diff
@@ -120,7 +120,7 @@ let ocamldep = get_prog bin_dir "ocamldep"
 let run_ocamllex src =
   let dst = String.sub src ~pos:0 ~len:(String.length src - 1) in
   let x = Sys.file_exists dst in
-  let n = exec "%s -q %s" ocamllex src in
+  let n = exec "%s -q %s" (Filename.quote ocamllex) src in
```
Comment (dra27) [inline bootstrap.ml]: You noted in https://github.com/janestreet/jbuilder/issues/322#issue-272748044 that quoting here wasn't working. Looking at the output of [this build](https://ci.appveyor.com/project/Chris00/ocaml-appveyor/build/1.0.190#L5118), the PATH in question included a forward-slash and you were also quoting `src`.

Did changing one of those two things fix the problem here? I'm happy to investigate this further, I'm just wondering if you already had?

Comment (Chris00) [reply bootstrap.ml]: Sorry, I did not check all combinations as I needed to rebuild the compiler each time — my interest was to have the compiler and jbuilder ASAP to test other packages.  I'll enable more recent versions of the compiler in my fork to test it more thoroughly — I doubt I can find the time to do that before this weekend though.

**Comment by dra27** [comment]
Thanks for this - forward-slashes and back-slashes definitely aren't the issue, it's having multiple quotes in the original example in #322 which was the problem there. That doesn't affect the bootstrap, as the arguments can never need quoting (since they're OCaml filenames).

This is therefore good to go.

**Pull request merged**